python/tvm/meta_schedule/tir_integration.py

```diff
@@ -58,6 +58,7 @@ def tune_tir(  # pylint: disable=too-many-locals
     seed: Optional[int] = None,
     module_equality: str = "structural",
     special_space: Optional[Mapping[str, SpaceGenerator.SpaceGeneratorType]] = None,
+    **kwargs,
```

Comment:
May I ask if this is specifically added for cost model? If so, IMHO it would be better to avoid confusion and only add one parameter in cost model's constructor function instead.

Reply:
Hi zhou, thanks for your time to review, there are some other cost models and set_tree_method only work for xgb model，so I think it is not a proper way to specialize the parameter here. And there may be some other parameters need to be set in other cases, use **kwargs may be more flexible.

Reply:
Hi Vincent, thanks for the explanation, it makes sense to me that `set_tree_method` only work for xgb model and we shouldn't specialize it for all the cost models. On the other hand, the `**kwargs` here is used in a `tune_tir` api and it would be a bit hard to figure out if certain argument is given for cost model or database, or even other classes. Another issue is when multiple class use the same argument, it would be hard to handle. IMHO we may try to let the users configure cost model separately and call in the `tune_tir` api as follows:
```python
cost_model = CostModel.create("xgb", num_tuning_cores=num_cores, tree_method = "auto")
tune_tir(mod, ..., cost_model=cost_model)
```
What do you think?

Reply:
Thanks for your time! I‘m agree with your proposal, I will fix it.